core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -211,6 +211,11 @@ public long getSizeInBytes()
                 (rleValue != null ? rleValue.getSizeInBytes() : 0);
     }
 
+    public long getDirectSizeInBytes()
+    {
+        return rleValue == null ? getSizeInBytes() : (rleValue.getSizeInBytes() * rlePositionCount);
```

Comment:
`getSizeInBytes()` is not the right call if dictionary is being build. Could you add a comment (if we don't plan to fix it for now?)

Reply:
Added a clarifying comment, similar to the one in `getSizeInBytes()`